### ISSUE TITLE
xdg-open unexpected argument '/c'

### ISSUE THREAD

**Issue by spaasis** [OmniSharp/omnisharp-vscode#3060]
## Issue Description ##
An error is thrown when loading an ASP.NET Core project. Other project types (Standard, Core) seem to work.

## Steps to Reproduce ##
Open a solution containing an ASP.NET core project in VSCode (only tested on one project)

## Expected Behavior ##
Project would load correctly

## Actual Behavior ##
An error is thrown early on:
```
<details>[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: /path/to/project.csproj
xdg-open: unexpected argument '/c'
Try 'xdg-open --help' for more information.
[fail]: OmniSharp.MSBuild.ProjectLoader
        Cannot find the specified file
```
### OmniSharp log ###
<details>
Starting OmniSharp server at 2019-5-12 09:48:10
    Target: /path/to/solution.sln

OmniSharp server started with Mono 5.20.1.
    Path: /home/s/.vscode/extensions/ms-vscode.csharp-1.19.1/.omnisharp/1.32.19/omnisharp/OmniSharp.exe
    PID: 21793

[info]: OmniSharp.Stdio.Host
        Starting OmniSharp on antergos 0.0 (x64)
[info]: OmniSharp.Services.DotNetCliService
        DotNetPath set to dotnet
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Located 2 MSBuild instance(s)
            1: Mono 15.0 - "/usr/lib/mono/msbuild/15.0/bin"
            2: StandAlone 15.0 - "/home/s/.vscode/extensions/ms-vscode.csharp-1.19.1/.omnisharp/1.32.19/omnisharp/.msbuild/Current/Bin"
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Registered MSBuild instance: Mono 15.0 - "/usr/lib/mono/msbuild/15.0/bin"
            CscToolExe = csc.exe
            CscToolPath = /home/s/.vscode/extensions/ms-vscode.csharp-1.19.1/.omnisharp/1.32.19/omnisharp/.msbuild/Current/Bin/Roslyn
[info]: OmniSharp.Cake.CakeProjectSystem
        Detecting Cake files in '/path/to/repo'.
[info]: OmniSharp.Cake.CakeProjectSystem
        Could not find any Cake files
[info]: OmniSharp.WorkspaceInitializer
        Project system 'OmniSharp.DotNet.DotNetProjectSystem' is disabled in the configuration.
[info]: OmniSharp.MSBuild.ProjectSystem
        Detecting projects in '/path/to/solution.sln'.
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for ...
[info]: OmniSharp.Script.ScriptProjectSystem
        Detecting CSX files in '/path/to/repo'.
[info]: OmniSharp.Script.ScriptProjectSystem
        Could not find any CSX files
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.CSharpWorkspaceOptionsProvider
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: /path/to/working.csproj
[info]: OmniSharp.WorkspaceInitializer
        Configuration finished.
[info]: OmniSharp.Stdio.Host
        Omnisharp server running using Stdio at location '/path/to/repo/' on host 21640.
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file '/path/to/working.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Adding project '/path/to/working.csproj''
... more working projects
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: '/path/to/aspnet.csproj'
xdg-open: unexpected argument '/c'
Try 'xdg-open --help' for more information.
[fail]: OmniSharp.MSBuild.ProjectLoader
        Cannot find the specified file
[warn]: OmniSharp.MSBuild.ProjectManager
        Failed to load project file '/path/to/aspnet.csproj'
/path/to/aspnet.csproj
/home/s/.nuget/packages/buildwebcompiler/1.12.394/build/BuildWebCompiler.targets(12,9): Error: Cannot find the specified file

</details>

## Environment information ##

**VSCode version**: 1.33.1
**C# Extension**: 1.19.1

<details><summary>Mono Information</summary>
        OmniSharp using global mono :5.20.1</details>
<details><summary>Dotnet Information</summary>
.NET Core SDK (reflecting any global.json):
 Version:   2.2.105
 Commit:    7cecb35b92

Runtime Environment:
 OS Name:     antergos
 OS Version:  
 OS Platform: Linux
 RID:         arch-x64
 Base Path:   /opt/dotnet/sdk/2.2.105/

Host (useful for support):
  Version: 2.2.3
  Commit:  6b8ad509b6

.NET Core SDKs installed:
  2.2.105 [/opt/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.NETCore.App 2.2.3 [/opt/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|
|---|---|---|
|azure-account|ms-vscode|0.8.2|
|azure-pipelines|ms-azure-devops|1.147.2|
|azurerm-vscode-tools|msazurermtools|0.6.0|
|csharp|ms-vscode|1.19.1|
|Ionide-fsharp|Ionide|3.37.0|
|Ionide-Paket|Ionide|1.12.0|
|mssql|ms-mssql|1.6.0|
|team|ms-vsts|1.149.2|
|vscode-azureappservice|ms-azuretools|0.14.0|
|vscode-azurestorage|ms-azuretools|0.7.0|
|vscode-node-azure-pack|ms-vscode|0.0.6|
|vscode-nuget-package-manager|jmrog|1.1.6|;

</details>

**Comment by gregg-miskelly** [comment]
@spaasis `xdg-open` is used for launching a browser. You can likely fix this either by disabling browser launch, or by changing the command line. To do either, open launch.json, and customize the 'launchBrowser' element.

**Comment by gregg-miskelly** [comment]
Sorry, upon rereading this issue, I realized that I completely misread this bug. Ignore my previous comment.

**Comment by rmunn** [comment]
Possibly related, depending on where that `/c` parameter is ultimately coming from, to https://github.com/dotnet/corefx/issues/24704, where `xdg-open` was being used on Linux to provide the equivalent of `cmd /c`. If the process of opening a `.csproj` file involves starting an external process, then I'd strongly suspect that https://github.com/dotnet/corefx/issues/24704 is behind the issue. It would be helpful to figure out the full `xdg-open` command that's being attempted here.

**Comment by spaasis** [comment]
Is there something I can do to help you guys catch this?

**Comment by gregg-miskelly** [comment]
CC @rchande @akshita31

**Comment by spaasis** [comment]
Hmm, if I uninstall the WebCompiler (https://github.com/madskristensen/WebCompiler) and restart Omnisharp there's no errors, so I guess this is actually a WebCompiler bug. Should I close this one when I post a new issue over there?

**Comment by spaasis** [comment]
Seems to work fine when I use VSCode on Windows, so it's probably a Linux bug,

**Comment by rmunn** [comment]
As I mentioned in https://github.com/madskristensen/WebCompiler/issues/421#issuecomment-503220197, WebCompiler seems to assume it's running on Windows (calling `cmd.exe /c` and so on), so it's not designed to work on Linux.

@spaasis I'm curious to know how you're managing to use WebCompiler in VS Code, as it's a Visual Studio extension, not a VS Code extension. Is it a NuGet package you use in your project? When you said that you "uninstalled the WebCompiler" and it worked, what exactly did you do?

**Comment by spaasis** [comment]
@rmunn Yup, that seems to close this case.

So, to clarify;
I was working on a project on Windows, Visual Studio, that utilizes the WebCompiler extension. That extensions seems to automatically install a NuGet package BuildWebCompiler (https://www.nuget.org/packages/BuildWebCompiler) that does.. something (although earlier today I ran a different WebCompiler project without the NuGet package and it compiled the .scss just fine, go figure..).

I tried to open this project on my Linux machine with VSCode and this error popped up. When I said I 'uninstalled WebCompiler', I meant that I removed that BuildWebCompiler NuGet package, nothing fancier than that :)
After removing it, the Omnisharp proceeded without errors, and that's as far as I went with testing.

Thank you guys for helping me out here. I'll close the issue and continue discussing it over at the WebCompiler side.
